### ISSUE TITLE
Generating an URL from non sanitized request parameters is insecure

### ISSUE THREAD

**Issue by Jonjoe** [Sology/smart_listing#79]
# The Problem

Basic usage in Rails 5.0.0.beta2 results with the above. 

#### Model
```
class CreateStudents < ActiveRecord::Migration[5.0]
  def change
    create_table :students do |t|
      t.string :first_name
      t.string :last_name
      t.belongs_to :teacher
      t.timestamps
    end
  end
end
```
#### Controller
```
class HomeController < ApplicationController

  include SmartListing::Helper::ControllerExtensions
  helper  SmartListing::Helper

  def index
    @students = smart_listing_create(:students, Student.all, partial: "students/listing")
  end

end
```
#### View Parent
```
<h1>Students</h1>
<%= smart_listing_render :students %>
```
#### View Child
```
<table>
  <thead>
    <th><%= smart_listing.sortable "First Name", :first_name %></th>
    <th><%= smart_listing.sortable "Last Name", :last_name %></th>
  </thead>
  <tbody>
    <%= @students.each do |student| %>
      <tr>
        <td><%= student.first_name %></td>
        <td><%= student.last_name %></td>
      </tr>
    <% end %>
  </tbody>
</table>
<%= smart_listing.paginate  %>
```

# The work around
Found this: [stackoverflow question](http://stackoverflow.com/questions/34413971/rails-5-0-0-beta1-generating-an-url-from-non-sanitized-request-parameters-is-i)

tl:dr; add this to production.rb and development.rb 
```
config.action_controller.permit_all_parameters = true
```
This is not ideal as you are bypassing security measures ... hence posting this issue :P.

**Comment by rexmadden** [comment]
According to one of the answers - http://stackoverflow.com/a/34437768/3597961 - the latest version of kaminari fixes it.

**Comment by rush-skills** [comment]
I tried using the latest version of kaminari (0.17) in Rails 5 but it still didn't fix the problem, as it relates to params from smart_listing itself. Also, since the current release has kaminari locked to 0.16, it isn't possible to directly use it in the Gemfile.

**Comment by ljachymczyk** [comment]
I think this is already fixed by #92
